doc/stream-telemetry/stream-telemetry-hld.md

```diff
@@ -0,0 +1,930 @@
+      - [Config DB Enhancements](#config-db-enhancements)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Memory Consumption](#memory-consumption)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+
+
+## Table of Content 
+
+### Revision  
+
+### Scope  
+
+This section describes the scope of this high-level design document in SONiC.
+
+### Definitions/Abbreviations 
+
+This section covers the abbreviation if any, used in this high-level design document and its definitions.
+
+### Overview 
+
+The purpose of this section is to give an overview of high-level design document and its architecture implementation in SONiC. 
+
+### Requirements
+
+This section list out all the requirements for the HLD coverage and exemptions (not supported) if any for this design.
+
+### Architecture Design 
+
+This section covers the changes that are required in the SONiC architecture. In general, it is expected that the current architecture is not changed.
+This section should explain how the new feature/enhancement (module/sub-module) fits in the existing architecture. 
+
+If this feature is a SONiC Application Extension mention which changes (if any) needed in the Application Extension infrastructure to support new feature.
+
+``` mermaid
+
+---
+title: Stream telemetry architecture
+---
+flowchart BT
+    subgraph Redis
+        config_db[(CONFIG_DB)]
+        state_db[(STATE_DB)]
+    end
+
+    subgraph SONiC service
+    subgraph GNMI container
+        gnmi(GNMI server)
+        counter_syncd(Counter Syncd)
+    end
+    subgraph SWSS container
+    subgraph Orchagent
+        st_orch(Stream Telemetry Orch)
+    end
+    end
+    subgraph SYNCD container
+        syncd(Syncd)
+    end
+    end
+
+    subgraph Linux Kernel
+        dma_engine(DMA Engine)
+        netlink_module(Netlink Module)
+    end
+
+    asic[\ASIC\]
+
+    config_db --STREAM_TELEMETRY_PROFILE
+                STREAM_TELEMETRY_GROUP--> st_orch
+    st_orch --SAI_OBJECT_TYPE_TAM_REPORT
+                SAI_OBJECT_TYPE_TAM_TEL_TYPE
+                SAI_OBJECT_TYPE_TAM_TRANSPORT
+                SAI_OBJECT_TYPE_TAM_COLLECTOR
+                SAI_OBJECT_TYPE_TAM_EVENT
+                SAI_OBJECT_TYPE_TAM--> syncd
+    syncd --IPFIX template--> st_orch
+    syncd --TAM configuration--> dma_engine
+    syncd --TAM configuration--> netlink_module
+    st_orch --STREAM_TELEMETRY_SESSION--> state_db
+    state_db --STREAM_TELEMETRY_SESSION--> counter_syncd
+    asic --counters--> dma_engine
+    dma_engine --IPFIX record--> netlink_module
+    netlink_module --IPFIX record--> counter_syncd
+    counter_syncd --telemetry message--> gnmi
+```
+
+### High-Level Design ###
+
+#### Modules ####
+
+##### Netlink Module #####
```

Comment:
i know the numbered prefix are not there in the template, but the doc will be much more readable with it. we should consider add them here.